vm/actor/tests/paych_actor_test.rs

```diff
@@ -0,0 +1,1192 @@
+        )
+        .is_ok());
+    rt.verify();
+
+    let state: PState = rt.get_state().unwrap();
+    let ls_updated: &LaneState = &state.lane_states[1];
+    let big_delta =
+        &sv.amount - BigInt::from_signed_bytes_be(&ls_to_update.redeemed.to_signed_bytes_be());
+
+    let exp_send = big_delta + BigInt::from_signed_bytes_be(&initial_amount.to_radix_be(10));
+    assert_eq!(
+        exp_send,
+        BigInt::from_radix_be(Sign::Plus, &state.to_send.to_radix_be(10), 10).unwrap()
+    );
+    assert_eq!(sv.amount, ls_updated.redeemed);
+    assert_eq!(sv.nonce, ls_updated.nonce);
+}
+
+#[test]
+fn merge_success() {
+    let num_lanes = 3;
+    let bs = MemoryDB::default();
+    let (mut rt, mut sv) = require_create_cannel_with_lanes(&bs, num_lanes);
+    let state: PState = rt.get_state().unwrap();
+    let state_2: PState = rt.get_state().unwrap();
+
+    rt.set_caller(ACCOUNT_ACTOR_CODE_ID.clone(), state.from);
+    let merge_to: &LaneState = &state.lane_states[0];
+    let merge_from: &LaneState = &state.lane_states[1];
+    sv.lane = merge_to.id;
+    let merge_nonce = merge_to.nonce + 10;
+    let merges: Vec<Merge> = vec![Merge {
+        lane: merge_from.id,
+        nonce: merge_nonce,
+    }];
+
+    sv.merges = merges;
+    let payee_addr = Address::new_id(103);
+    rt.expect_validate_caller_addr(&[state.from, state.to]);
+    rt.expect_verify_signature(
+        sv.clone().signature.unwrap(),
+        payee_addr,
+        to_vec(&sv).unwrap(),
+        ExitCode::Ok,
+    );
+    let ucp = UpdateChannelStateParams {
+        proof: vec![],
+        secret: vec![],
+        sv: sv.clone(),
+    };
+    assert!(rt
+        .call(
+            &PAYCH_ACTOR_CODE_ID.clone(),
+            Method::UpdateChannelState as u64,
+            &Serialized::serialize(ucp).unwrap()
+        )
+        .is_ok());
+    rt.verify();
+
+    let exp_merge_to = LaneState {
+        id: merge_to.id,
+        redeemed: sv.amount.clone(),
+        nonce: sv.nonce,
+    };
+
+    let exp_merge_from = LaneState {
+        id: merge_from.id,
+        redeemed: merge_from.redeemed.clone(),
+        nonce: merge_nonce,
+    };
+
+    let redeemed = &merge_from.redeemed + &merge_to.redeemed;
+
+    let exp_delta = &sv.amount - &redeemed;
+
+    let mut exp_state = state_2;
+
+    exp_state.to_send = exp_delta.to_biguint().unwrap() + &state.to_send;
+
+    exp_state.lane_states = vec![
+        exp_merge_to,
+        exp_merge_from,
+        exp_state.lane_states.pop().unwrap(),
+    ];
+    verify_state(&mut rt, num_lanes as i64, exp_state);
+}
+
+#[test]
+fn merge_failue() {
+    let lane_vec = vec![1, 1, 1, 0];
+    let voucher_vec = vec![10, 0, 10, 10];
+    let balance_vec = vec![0, 0, 1, 0];
+    let merge_vec = vec![1, 10, 10, 10];
+    let exit_vec = vec![
+        ExitCode::ErrIllegalArgument,
+        ExitCode::ErrIllegalArgument,
+        ExitCode::ErrIllegalState,
+        ExitCode::ErrIllegalArgument,
+    ];
```

Comment:
Why not just create a struct and have a 4 len array of them so you don't have to index each like this

Reply:
Yeah idk what I was thinking tbh it wouldve been less work to put in a actual struct